### ISSUE TITLE
Run multiple small example apps with react-scripts

### ISSUE THREAD

**Issue by ashander** [gvwilson/js-vs-rc#3]
As motivated in this tweet, https://twitter.com/gvwilson/status/868462924419002368  Some potential solutions are discussed in the replies.

**Comment by gvwilson** [comment]
Creating one target in `package.json` for each tutorial mini-app is a possibility - is there a way to reference extra command-line arguments in script descriptions in `package.json` so that:

    npm start abc

will `cd ./src/react/abc && react-scripts start` ?

**Comment by gvwilson** [comment]
Experiment: `package.json` contains:

```
  "scripts": {
    "demo": "cd foo && npm start",
    "start": "react-scripts start",
    ...
  }
```

and I have made `./foo/public/index.html` and `./foo/src/index.js`.  When I execute `npm run demo`, the server starts, but is serving from `./public/ and `./src`, rather than `./foo/*`, so it looks like the `npm start` command is looking upward to find the `package.json` and then executing using that directory as its root.

**Comment by gvwilson** [comment]
Digging deeper into `webpack-dev-server` and its configuration... maze of twisty little paths, all alike...

**Comment by ashander** [comment]
I mentioned this elsewhere but will add here:

> it's clear that react-scripts is pretty inflexible https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md#folder-structure. 

with a little more digging it seems clearer that other solutions aren't going to work easily in react-app proper (cf https://github.com/facebookincubator/create-react-app/issues/1084) but using `eject` and configuring [webpack entry points](https://webpack.js.org/concepts/entry-points/#components/sidebar/sidebar.jsx) might work.

But, as I mentioned 

>I'm not a node master by any means, but from what I understand of the node way it often involves many redundant node_modules across small apps or modules. So maybe that would be the easiest solution?

To unpack the "that" quoted here --- I mean writing each small app as a self-contained module. Running and installing these would of course result in many duplicate `node_modules` and wasted network usage, but it might be simpler.

**Comment by gvwilson** [comment]
I have a good-enough solution via https://www.sitepoint.com/getting-started-react-jsx/:

1. `npm install --save http-server`
2. `mkdir foo`
3. create `foo/index.html` and `foo/app.js` with the content shown below
4. in `./foo`, run `../node_modules/.bin/http-server`
5. Go to `localhost:8080`

The little server whines about not finding `favicon.ico`, but that's easily fixed.

`index.html`
```
<!DOCTYPE html>
<html>
  <head>
    <title>Hello World</title>
    <meta charset="utf-8">
    <script src="https://fb.me/react-15.0.1.js"></script>
    <script src="https://fb.me/react-dom-15.0.1.js"></script>
    <script src="https://unpkg.com/babel-standalone@6/babel.js"></script>
    <script src="app.js" type="text/babel"></script>
  </head>
  <body>
    <div id="app">
      <!-- this is filled in -->
    </div>
  </body>
</html>
```

`app.js`
```
ReactDOM.render(
  <h1>Hello, world</h1>,
  document.getElementById("app")
)
```

**Comment by gvwilson** [comment]
Solved (enough) in 60a24fa694 - see comments on that commit for plan.
